plonky2x/core/src/frontend/eth/rlp/decoder.rs

```diff
@@ -0,0 +1,249 @@
+//! This file implements RLP decoder.
+//!
+//! Reference: https://ethereum.org/en/developers/docs/data-structures-and-encoding/rlp/
+//!
+//! We have a total of 6 different ways of representing Merkle Patricia Tries.
+//! - Encoded versions
+//!     1. Encoded MPT : vec<u8>
+//!     2. Padded Encoded MPT : ([u8; usize], usize)
+//!     3. Circuita Padded Encoded MPT : ArrayVariable<ByteVariable, ENCODING_LEN>
+//! - Decoded versions
+//!     1'. MPT : RLPItem
+//!     2'. Padded MPT : MPTNodeFixedSize
+//!     3'. Circuit Padded MPT : ArrayVariable<ArrayVariable<ByteVariable, 32>, 17>
```

Comment:
We can make a new type here called like MPTFixedSizeVariable

Reply:
Oh okay! I like that. Maybe in the next PR?